torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -634,6 +638,10 @@ def _sharded_pre_load_state_dict_hook(
                 tensor, local_tensor, group=fsdp_state.process_group
             )
         tensor = tensor.narrow(0, 0, param_numel).reshape(param.size())
+        if fqn_to_param_ext.get(fqn) is not None:
```

Comment:
Curious: When do we expect `fqn_to_param_ext.get(fqn)` to not have `fqn` in `fqn_to_param_ext` and hence return the default `get()` value of `None` (i.e. not because the element in `_param_extenssions` is `None`)?

Reply:
Yup, I think that's when only FSDP is applied, not 2D's case

Reply:
This happens when the FQN is for a regular tensor and not a DTensor.